jvm/src/Language/Java.hs

```diff
@@ -712,6 +712,24 @@ withStatic [d|
 
   instance Reflect (Vector Int32) where
     reflect = reflect <=< Vector.thaw
+
+  instance Interpretation (IOVector Double) where
+    type Interp (IOVector Double) = 'Array ('Prim "double")
+
+  instance Reify (IOVector Double) where
+    reify = reifyMVector (getDoubleArrayElements) (releaseDoubleArrayElements)
+
+  instance Reflect (IOVector Double) where
+    reflect = reflectMVector (newDoubleArray) (setDoubleArrayRegion)
+
+  instance Interpretation (Vector Double) where
+    type Interp (Vector Double) = 'Array ('Prim "double")
+
+  instance Reify (Vector Double) where
+    reify = Vector.freeze <=< reify
+
+  instance Reflect (Vector Double) where
+    reflect = reflect <=< Vector.thaw
```

Comment:
How about the more general
```Haskell
  instance Interpretation (VS.Vector a) where
    type Interp (VS.Vector a) = Interp (IOVector a)

  instance (Storable a, Reify (IOVector a)) => Reify (VS.Vector a) where
    reify = VS.freeze <=< reify

  instance (VS.Storable a, Reflect (IOVector a)) => Reflect (VS.Vector a) where
    reflect = reflect <=< VS.unsafeThaw
```
?

We have this in a fork and haven't had the time to port it yet.

Reply:
Yes, good idea

Reply:
I've pushed another commit that implements this.

Reply:
re safe/unsafe thawing – caught my eye as well but I'd rather not lump that change into this PR